src/common/transformations/src/transformations/common_optimizations/mul_conv_fusion.cpp

```diff
@@ -106,7 +111,7 @@ ov::pass::MultiplyGroupConvolutionFusion::MultiplyGroupConvolutionFusion() {
             // Reshape mul_const from shape (1, C, H, W) to (G, 1, C / G, H, W) to match GroupConvolution weights format
             Shape new_shape{G, 1, C};
             std::copy(mul_const_shape.begin() + 2, mul_const_shape.end(), std::back_inserter(new_shape));
-            if (op::util::check_for_broadcast(weights_shape, new_shape)) {
+            if (!ov::op::util::broadcasted_only_channel(weights_shape, new_shape)) {
```

Comment:
it looks like, for Mul -> Conv case we have to check not only the channel dim
so please revert the changes